dashboard/app/models/course_version.rb

```diff
@@ -130,4 +137,32 @@ def self.course_offering_keys(content_root_type)
       CourseVersion.includes(:course_offering).where(content_root_type: content_root_type).map {|cv| cv.course_offering&.key}.compact.uniq.sort
     end
   end
+
+  def recommended?(locale_code = 'en-us')
+    return false unless stable?
+    return true if course_offering.course_versions.length == 1
+
+    family_name = course_offering.key
+    latest_stable_version = content_root_type == 'UnitGroup' ? UnitGroup.latest_stable_version(family_name) : Script.latest_stable_version(family_name, locale: locale_code)
+
+    latest_stable_version == content_root
+  end
+
+  def summarize_for_assignment_dropdown(user, locale_code)
+    [
+      id,
+      {
+        id: id,
+        version_year: key,
+        content_root_id: content_root.id,
+        name: content_root.localized_title,
+        path: content_root.link,
+        type: content_root_type,
+        is_stable: stable?,
+        is_recommended: recommended?(locale_code),
+        locales: content_root_type == 'UnitGroup' ? ['English'] : content_root.supported_locale_names,
```

Comment:
I think so. Currently there is nothing on unit_group like supported_locales on script. I don't know if there are plans to add it soon since we are working on CSD. Maybe @dju90 would know.